### PR TITLE
Add mac release workflow

### PR DESCRIPTION
**Pull request by OKendigelyan**
## Proposed changes

This PR adds mac release workflow

### CODE REVIEW

**Review by ajinkyaraj-23 — APPROVED** [review]
LGTM

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -21,36 +34,72 @@ jobs:
...
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20.18.0
+          cache: "pnpm"
+
+      - name: Install dependencies
+        run: pnpm install
+
+      - name: Build
+        run: pnpm build
```
Comment (serjonya-trili) [inline .github/workflows/release.yaml:94]: you're building all the projects which is redundant. it's better to run turbo build with `working-directory: apps/desktop`

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -1,17 +1,30 @@
...
 on:
   push:
-     tags:
-       - 'v*.*.*'
+    tags:
+      - "v*.*.*"
+
 jobs:
-  create-release-linux:
+  create-draft-release:
```
Comment (serjonya-trili) [inline .github/workflows/release.yaml:9]: it's better to make the downstream jobs explicitly depend on this one

File: .github/workflows/release.yaml (hunk 1 above)
Comment (ajinkyaraj-23) [reply .github/workflows/release.yaml:94]: Yes. Thats right.

Comment (ajinkyaraj-23) [reply .github/workflows/release.yaml:95]: It does not work. Need to build the dependency projects also. need to run it from root directory.

File: .github/workflows/release.yaml (hunk 2 above)
Comment (ajinkyaraj-23) [reply .github/workflows/release.yaml:9]: Its already dependent. see the needs: tag in below jobs

**Pull request merged**